OSDOCS-18925:Docs updates for [OSD-GCP] allowing for edit of ingress controller to use namespaceSelector. (#109873)
openshift/openshift-docs: modules/osd-create-cluster-exclude-namespace-selector-cli.adoc

// Module included in the following assemblies:
//
// * networking/networking_operators/ingress-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="osd-create-cluster-exclude-namespace-selector-cli_{context}"]
= Setting namespace exclusions for the default ingress when creating a cluster

[role="_abstract"]

> When you create an {product-title} cluster in noninteractive mode, you can pass a namespace label selector so that namespaces matching those labels are excluded from the default `application ingress.` This allows you to exclude namespaces that host workloadsthrough the default ingress, such as namespaces with sensitive data or internal services.

Reviewer: ```suggestion
When you create an {product-title} cluster in non-interactive mode, you can pass a namespace label selector so that namespaces matching those labels are excluded from the default `application ingress.` This allows you to exclude namespaces that host workloads through the default ingress, such as namespaces with sensitive data or internal services.
```
found typo

[IMPORTANT]
====
Do not exclude namespaces that host required platform routes (for example, `openshift-console` or `openshift-authentication`). Excluding them can break the web console, downloads, or OAuth flows.
====

.Procedure

. Run `ocm create cluster -h` and confirm that your `ocm` version lists the `--exclude-namespace-selector` flag.

. Build your `ocm create cluster` command with the required parameters for your cloud provider and subscription model. Add `--exclude-namespace-selector` with a comma-separated list of `key=value` pairs. Do not include spaces around the `=` sign.
+
[NOTE]
====
To exclude namespaces by metadata name only (a comma-separated list of namespace names without label keys), use `--default-ingress-excluded-namespaces` instead. Run `ocm create cluster -h` for the current description of both flags.
====
+
The following example shows only the ingress-related fragment; replace the rest of the flags with the values required for your environment.
+
[source,terminal]
----
$ ocm create cluster <cluster_name> \
  --provider=<aws_or_gcp> \
  <other_required_flags> \

>   --exclude-namespace-selector '<key>=<value>,<key2>=<value2>'

Reviewer: ```suggestion
  --default-ingress-excluded-namespace-selectors '<key>=<value>,<key2>=<value2>'
```
flag name

----
+
where:

`<cluster_name>`:: Specifies the cluster name.

`--provider=<aws_or_gcp>`:: Specifies the cloud provider.

`<other_required_flags>`:: Required parameters such as region, version, CCS settings, or billing flags, as described in the cluster creation documentation for your platform.

> `--exclude-namespace-selector`:: Specifies label selectors; namespaces whose labels match are excluded from the default application ingress, subject to validation by the service. Replace `<key>=<value>` with your labels.

Reviewer: ```suggestion
`--default-ingress-excluded-namespace-selectors`:: Specifies label selectors; namespaces whose labels match are excluded from the default application ingress, subject to validation by the service. Replace `<key>=<value>` with your labels.
```

// Module included in the following assemblies:
//
// * networking/networking_operators/ingress-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="osd-create-cluster-exclude-namespace-selector-cli_{context}"]
= Set namespace exclusions for the default ingress when creating a cluster

[role="_abstract"]

> When you create an {product-title} cluster in noninteractive mode, you can pass a namespace label selector so that namespaces matching those labels are excluded from the default `application ingress.` This allows you to exclude namespaces that host workloads through the default ingress, such as namespaces with sensitive data or internal services.

Reviewer 2: ```suggestion
When you create an {product-title} cluster in noninteractive mode, you can pass a namespace label selector so that namespaces matching those labels are excluded from the default `application ingress`. This allows you to exclude namespaces that host workloads through the default ingress, such as namespaces with sensitive data or internal services.
```

.Prerequisites

* You installed the `ocm` CLI and logged in with credentials that can create clusters in {cluster-manager-first}.
* You are using the noninteractive mode for `ocm create cluster`. For interactive mode, use the prompts for ingress settings when they are available for your `ocm` version.

> [IMPORTANT]

Reviewer 2: Just an informational note: I'm not sure how this admonition will translate to DITA. I'm guessing that it will be placed in a `<context>` element, if that makes a difference for you.

====
+
The following example shows only the ingress-related fragment. Replace the rest of the flags with the values required for your environment.
+
[source,terminal]
----
$ ocm create cluster <cluster_name> \
  --provider=<aws_or_gcp> \
  <other_required_flags> \
  --default-ingress-excluded-namespace-selectors '<key>=<value>,<key2>=<value2>'
----
+
where:

`<cluster_name>`:: Specifies the cluster name.

`--provider=<aws_or_gcp>`:: Specifies the cloud provider.

`<other_required_flags>`:: Required parameters such as region, version, CCS settings, or billing flags, as described in the cluster creation documentation for your platform.

`--default-ingress-excluded-namespace-selectors`:: Specifies label selectors; namespaces whose labels match are excluded from the default application ingress, subject to validation by the service. Replace `<key>=<value>` with your labels. Do not include spaces around the `=` sign.

.Verification

* After the cluster reaches `ready` state, confirm ingress settings and inspect the default ingress object for the configured exclusion data.
+
[source,terminal]
----

> $ ocm list ingress -c <cluster_name> -o json

Reviewer: ```suggestion
$ ocm list ingress -c <cluster_name> 
```

-o flag does not exist for this command.
